spring-kafka/src/main/java/org/springframework/kafka/listener/ConcurrentMessageListenerContainer.java

```diff
@@ -259,6 +271,10 @@ protected void doStop() {
 				container.stop();
 			}
 			this.containers.clear();
+			if (this.usingDefaultListenerExecutor) {
+				setListenerTaskExecutor(null); // in case the concurrency changes before the next start
+				this.usingDefaultListenerExecutor = false;
```

Comment:
Shouldn't we `ThreadPoolTaskExecutor.shutdown()` in case of `if (this.usingDefaultListenerExecutor)` ?